File: questions/17_anagram_similarity.rb
# this is a multi-line comment in ruby
=begin
https://www.hackerrank.com/challenges/ctci-making-anagrams
=end

a = gets.strip
b = gets.strip

freq_count_a = a.split('').group_by{ |a| a[0] }.map {|k, v| [k, v.count]}.to_h
count = 0
b.split('').each do |k|
  if freq_count_a[k] && freq_count_a[k] > 0
    freq_count_a[k] -= 1
  else
    count += 1
  end
end

count += freq_count_a.values.sum

puts count

File: questions/17_anagram_similarity.cpp
/*
https://www.hackerrank.com/challenges/ctci-making-anagrams
*/

#include <iostream>

using namespace std;

#define REP(i, n) for(int i = 0; i < n; ++i)

int deletions_needed(string a, string b) {

  int A[26] = {0};
  int index;

  REP(i, a.length()) {
    index = a[i] - 'a';
    ++A[index];
  }

  int count = 0;
  REP(i, b.length()) {
    index = b[i] - 'a';
    if (A[index] > 0)
      --A[index];
    else
      ++count;
  }

  REP(i, 26)
    count += A[i];

  return count;
}

int main() {
  string a, b;

  cin >> a >> b;
  cout << deletions_needed(a, b) << endl;

  return 0;
}